File: common/src/main/java/com/xpf/common/utils/LogUtils.java
package com.xpf.common.utils;

import android.util.Log;

import com.xpf.common.BuildConfig;

/**
 * Created by xpf on 2017/9/22 :)
 * Function: logger utils.
 */

public class LogUtils {

    // default open log print.
    private static boolean mDebug = BuildConfig.ENABLE_DEBUG;

    private LogUtils() {
    }

    public static void isDebug(boolean isDebug) {
        mDebug = isDebug;
    }

    public static void e(String TAG, String message) {
        if (mDebug) {
            Log.e(TAG, message);
        }
    }


    public static void i(String TAG, String message) {
        if (mDebug) {
            Log.i(TAG, message);
        }
    }

    public static void d(String TAG, String message) {
        if (mDebug) {
            Log.d(TAG, message);
        }
    }

    public static void v(String TAG, String message) {
        if (mDebug) {
            Log.v(TAG, message);
        }
    }

    public static void w(String TAG, String message) {
        if (mDebug) {
            Log.w(TAG, message);
        }
    }

    public static void wtf(String TAG, String message) {
        if (mDebug) {
            Log.wtf(TAG, message);
        }
    }
}

File: app/src/main/java/com/xpf/p2p/utils/LogUtils.kt
package com.xpf.p2p.utils

import android.util.Log
import com.xpf.p2p.BuildConfig

/**
 * Created by xpf on 2017/9/22 :)
 * Function: logger utils.
 */
object LogUtils {

    private var mDebug = BuildConfig.ENABLE_DEBUG

    @JvmStatic
    fun isDebug(isDebug: Boolean) {
        mDebug = isDebug
    }

    @JvmStatic
    fun e(TAG: String, message: String?) {
        if (mDebug) {
            Log.e(TAG, message ?: "null")
        }
    }

    @JvmStatic
    fun i(TAG: String, message: String?) {
        if (mDebug) {
            Log.i(TAG, message ?: "null")
        }
    }

    @JvmStatic
    fun d(TAG: String, message: String?) {
        if (mDebug) {
            Log.d(TAG, message ?: "null")
        }
    }

    @JvmStatic
    fun v(TAG: String, message: String?) {
        if (mDebug) {
            Log.v(TAG, message ?: "null")
        }
    }

    @JvmStatic
    fun w(TAG: String, message: String?) {
        if (mDebug) {
            Log.w(TAG, message ?: "null")
        }
    }

    @JvmStatic
    fun wtf(TAG: String, message: String?) {
        if (mDebug) {
            Log.wtf(TAG, message ?: "null")
        }
    }
}
